app/src/main/java/eu/kanade/tachiyomi/source/SourceManager.kt

```diff
@@ -89,7 +92,13 @@ class SourceManager(
     private fun registerStubSource(sourceData: SourceData) {
         scope.launch {
             val (id, lang, name) = sourceData
-            sourceRepository.upsertSourceData(id, lang, name)
+            val dbSourceData = sourceRepository.getSourceData(id)
+            if (dbSourceData != sourceData) {
```

Comment:
Is this check needed? The other check could just check if the source from the database is not null and check the names, no?

Reply:
I mean no need to upsert data in db if it's the same thing

Reply:
You could invert the statement and return early if that is the case to avoid the nesting

```suggestion
            if (dbSourceData == sourceData) return
```